hr_timesheet_overview/models/hr_employee_hour.py

```diff
@@ -0,0 +1,327 @@
+            }
+        return values
+
+    @api.model
+    def _get_attendances_values(self, employee, date_from):
+        """Retrieve attendances (from resource module) values mapped by date
+
+        These attendances are the matrix product of contractual hours by day
+        and week days (NOT the `hr.attendance` model).
+
+        Warning: it is intended to be injected in context to reduce calls
+
+        :param employee: an employee record
+        :param date_from: a datetime.date object (default first contract date, included)
+        :param date_to: a datetime.date object (default today, included)
+        """
+        date_to = fields.Date.today()
+        values_by_date = {}
+        for contract in employee.contract_ids:
+            _logger.debug(
+                f"""process contract '{contract.name}'
+                from {contract.last_hours_report_date} to {date_to}"""
+            )
+            ranged_dates = generate_dates_from_range(
+                contract.last_hours_report_date, date_to
+            )
+            for date in ranged_dates:
+                if date_from <= date >= date_to:
+                    continue
+                values = self._prepare_attendance_value(contract, date)
+                if values:
+                    values_by_date[date] = values
+        return list(values_by_date.values())
+
+    @api.model
+    def _hook_get_type_from_timesheet(self, timesheet):
+        """Hook to allow proper override of type value
+        Called in `_prepare_timesheets_lines` method
+        """
+        return "leave" if timesheet.holiday_id else "timesheet"
+
+    @api.model
+    def _hook_get_name_from_timesheet(self, timesheet):
+        """Hook to allow proper override of name value
+        Called in `_prepare_timesheets_lines` method
+        """
+        return timesheet.account_id.name if timesheet.holiday_id else timesheet.name
+
+    def _prepare_timesheet_line(self, timesheet):
+        """Retrieve a timesheet line values
+
+        Day ratio quantity is a percentage of consumed hours relative to the
+        attendance day max quantity to make a full day (2h on 8h day is 25%).
+
+        If `attendances_by_date` is not set in context or date is not found,
+        default hours and day quantity are used (cf: `DEFAULT_TIME_QTY`)
+
+        :param employee: an employee record
+        :param date_from: a datetime.date object (default first contract date, included)
+        """
+        uom_hour = self.env.ref("uom.product_uom_hour")
+        project = timesheet.project_id
+        hours_qty = timesheet.unit_amount
+        if timesheet.product_uom_id != uom_hour:
+            hours_qty = timesheet.product_uom_id._compute_quantity(
+                timesheet.unit_amount, uom_hour, round=False
+            )
+
+        # TODO?
+        # If attendance hours qty is empty, it is an extra work day
+        days_qty = hours_qty / timesheet.employee_id.resource_calendar_id.hours_per_day
```

Comment:
```suggestion
        attendances_by_date = self.env.context.get("attendances_by_date", [])
        project = timesheet.project_id
        attendance = get_or_default(timesheet.date, attendances_by_date)
        hours_qty = timesheet.unit_amount
        if timesheet.product_uom_id != uom_hour:
            hours_qty = timesheet.product_uom_id._compute_quantity(
                timesheet.unit_amount, uom_hour, round=False
            )
        # Get the day filled ratio
        # If attendance hours qty is empty, it is an extra work day
        days_qty = hours_qty / (
            attendance["hours_qty"] or DEFAULT_TIME_QTY["hours_qty"]
        )
        # If hours filled the day then prefer to get ratio from the conf
        # To avoid setting a full day if conf says half one only
        if days_qty == 1:
            days_qty = attendance["days_qty"]
```
We want to take the real hours quantity by day by contract here. Otherwize hours calculations will be wrong.

Reply:
the real quantity is taken from the linked resource, IMO it wasn't correct to have global default values as there will be 50%, 70% contracts so the day will be less than 8 hours,
 and there is no attendance in timesheet computation  suggested part is obsolite now

Reply:
see https://github.com/OCA/hr/pull/1076#discussion_r812716042